src/forms/forms.response.ts

```diff
@@ -66,6 +66,38 @@ class Response {
     responseGroupId: number;
 }
 
+class subQuestion {
```

Comment:
class naming convention is PascalCase

Reply:
we also don't want to repeat a lot of code. This block of code is exactly the same as Question, without subQuestion

Easiest way is to use `Question extends SubQuestion` 


```
class SubQuestion {
    @ApiProperty({ example: 1 })
    id: number;

    @ApiProperty({ example: 1 })
    order: number;

    @ApiProperty()
    inputType: InputType;

    @ApiProperty({ example: "Changes to be made for the next sprint?" })
    text: string;

    @Optional()
    @ApiProperty({
        example:
            "Share your thoughts on what could be changed for the next sprint",
    })
    description: string;

    @ApiProperty({ example: false })
    answerRequired: boolean;

    @Optional()
    @ApiProperty({ example: null })
    multipleAllowed: boolean;

    @Optional()
    @ApiProperty({ example: null })
    optionGroup: OptionGroup;
}

class Question extends SubQuestion {
    @Optional()
    @ApiProperty({ isArray: true })
    subQuestions: SubQuestion;
}

```

Reply:
thank you for your insight. After doing some research around, I understand that a "sub-question" is considered a part of a "question". Therefore, it does seem more appropriate to have Question as the base class and SubQuestion as a subclass? Whoever when I tried this approach I get an error on can't access SubQuestion before initializing. I was thinking if it is better to make a base class for both? 
```
class Question {
    @ApiProperty({ example: 1 })
    id: number;

    @ApiProperty({ example: 1 })
    order: number;

    @ApiProperty()
    inputType: InputType;

    @ApiProperty({ example: "Changes to be made for the next sprint?" })
    text: string;

    @Optional()
    @ApiProperty({
        example:
            "Share your thoughts on what could be changed for the next sprint",
    })
    description: string;

    @ApiProperty({ example: false })
    answerRequired: boolean;

    @Optional()
    @ApiProperty({ example: null })
    multipleAllowed: boolean;

    @Optional()
    @ApiProperty({ example: null })
    optionGroup: OptionGroup;

    @Optional()
    @ApiProperty({ isArray: true })
    subQuestions?: SubQuestion[];
}

class SubQuestion extends Question {
    @Optional()
    @ApiProperty({ example: null })
    optionGroupId: number | null;

    @ApiProperty()
    parentQuestionId: number;

    @ApiProperty()
    createdAt: Date;

    @ApiProperty()
    updatedAt: Date;
}
```

Reply:
you can if you want to, but basically question should be the same as subquestion but with `subQuestions` field

so maybe you can do something like 

```
BaseQuestion {
  ...
}

Question extends BaseQuestions {
  subQuestions
}

SubQuestion extends BaseQuestions {}

```

if you want to include createdAT/updatedAt for subqestion we should have it for questions too
And you won't need "parentQuestionId" in subquestion as it would be a child node of a question. I'm not sure if we will ever need to pull a subquestion by itself

I think we will eventually need createdAt and updatedAt for both because it would be handy to show question last update in the admin dashboard